src/components/detail/detail.spec.js

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import { shallow, mount } from "enzyme";
+
+import Detail from "./detail.component";
+import {
+  elementsTagTest,
+  rootTagTest,
+} from "../../utils/helpers/tags/tags-specs";
+import { assertStyleMatch } from "../../__spec_helper__/test-utils";
+import {
+  StyledDetail,
+  StyledDetailContent,
+  StyledDetailIcon,
+  StyledDetailFootnote,
+} from "./detail.style";
+
+describe("Detail", () => {
+  let wrapper;
+
+  describe("render", () => {
+    beforeEach(() => {
+      wrapper = shallow(<Detail className="foo">foo</Detail>);
+    });
+
+    it("renders the children", () => {
+      expect(wrapper.find(StyledDetailContent).text()).toEqual("foo");
+    });
+
+    it("renders with custom classes", () => {
+      expect(wrapper.find(StyledDetail).props().className).toContain(
+        "carbon-detail foo"
+      );
+    });
+  });
+
+  describe("with a footnote", () => {
+    beforeEach(() => {
+      wrapper = shallow(<Detail footnote="extra info">foo</Detail>);
+    });
+
+    it("renders the footnote", () => {
+      const footnote = wrapper.find(StyledDetailFootnote);
+      expect(footnote.text()).toEqual("extra info");
+    });
+  });
+
+  describe("with an icon", () => {
+    beforeEach(() => {
+      wrapper = mount(<Detail icon="settings">foo</Detail>);
+    });
+
+    it("renders the icon", () => {
+      const icon = wrapper.find(StyledDetailIcon);
+      expect(icon).toBeDefined();
+    });
+
+    it("should give the content a margin left", () => {
+      assertStyleMatch(
+        { marginLeft: "26px" },
+        wrapper.find(StyledDetailContent)
+      );
+    });
+
+    describe("with a footnote", () => {
```

Comment:
Should this be "with a footnote and an icon"?

Reply:
No because it's wrapped in the `with an icon` describe block, so if you look at the test output, it's clear it has both: 
![image](https://user-images.githubusercontent.com/14963680/104181726-3247c280-5407-11eb-9b66-e75b26c9c896.png)